client/webserver/site/src/js/forms.ts

```diff
@@ -1848,6 +1852,71 @@ export class DepositAddress {
   }
 }
 
+// AppPassResetForm is used to reset the app apssword using the app seed.
+export class AppPassResetForm {
+  form: PageElement
+  page: Record<string, PageElement>
+  success: () => void
+
+  constructor (form: PageElement, success: () => void) {
+    this.form = form
+    this.success = success
+    const page = this.page = Doc.idDescendants(form)
+    bind(form, page.resetAppPWSubmitBtn, () => this.resetAppPW())
+  }
+
+  async resetAppPW () {
+    const page = this.page
+    const newAppPW = page.newAppPassword.value || ''
+    const confirmNewAppPW = page.confirmNewAppPassword.value
+    if (newAppPW === '') {
+      Doc.showFormError(page.appPWResetErrMsg, intl.prep(intl.ID_NO_PASS_ERROR_MSG))
+      return
+    }
+    if (newAppPW !== confirmNewAppPW) {
+      Doc.showFormError(page.appPWResetErrMsg, intl.prep(intl.ID_PASSWORD_NOT_MATCH))
+      return
+    }
+
+    // Clear the notification cache. Useful for development purposes, since
+    // the Application will only clear them on login, which would leave old
+    // browser-cached notifications in place after registering even if the
+    // client db is wiped.
+    app().setNotes([])
```

Comment:
Copy-pasta?

Reply:
Oh, yh. I now think there is no need to clear notes here, adds zero value.